keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
 
-```yaml
-apiVersion: cosi.io/v1alpha1
-kind: BucketAccessRequest
-metadata:
-  name:
-  namespace:
-  labels:
-    cosi.io/provisioner: [1]
-  finalizers:
-  - cosi.io/finalizer [2]
-spec:
-  serviceAccountName: [3]
-  bucketRequestName: [4]
-  bucketAccessClassName: [5]
-  bucketAccessName: [6]
-status:
-  accessGranted: [7]
-```
-
-1. `labels`: added by COSI.  Key’s value should be the provisioner name. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
-1. `finalizers`: added by COSI to defer `BucketAccessRequest` deletion until backend deletion ops succeed.
-1. `serviceAccountName`: (optional) the name of a Kubernetes ServiceAccount, in the same namespace, which is expected to be linked to an identity in the cloud provider, such that a pod using the ServiceAccount is authenticated as the linked identity. In this case, COSI applies the policy defined in the `BAC.policyActionsConfigMap` to the identity. If `serviceAccountName` is omitted, a "minted" user is generated by the driver and stored as `BucketAccess.principal`.
-1. `bucketRequestName`: (required) the name of the `BucketRequest` associated with this access request. From the bucket request, COSI knows the `Bucket` instance and thus the backend bucket and its properties.
-1. `bucketAccessClassName`: (required) name of the `BucketAccessClass` specifying the desired set of policy actions to be set for a user identity or ServiceAccount.
-1. `bucketAccessName`: name of the bound cluster-scoped `BucketAccess` instance. Set by COSI.
-1. `accessGranted`: if true access has been granted to the bucket. If false then access to the bucket has not been granted.
-
-#### BucketAccess
-
-A cluster-scoped administrative custom resource which encapsulates fields from the `BucketAccessRequest` (BAR) and the `BucketAccessClass` (BAC).  The purpose of the `BucketAccess` (BA) is to serve as an access communication path between provisioners and the central COSI controller.  The COSI controller creates a `BucketAccess` instance for a new `BucketAccessRequest`, and there is a 1:1 mapping between `BucketAccess` and `BucketAccessRequest`.
+1. The COSI ControllerManager is the central controller that manages lifecycle of COSI objects. Only one active instance of ControllerManager should be present in a cluster.
+2. The COSI Sidecar is the point of integration between COSI and drivers. All operations that require communication with the OSP is triggered by the Sidecar using gRPC calls to the driver. Only one active instance of Sidecar should be present **for a particular driver** in a cluster.
+3. The COSI driver communicates with the OSP to conduct Bucket related operations.
+4. The COSI NodeAdapter is responsible for presenting buckets and its associated access credentials to pods. One instance of NodeAdapter should be present on every node.
```

Comment:
"... on every node where a bucket workload can be scheduled."

Reply:
I think that is implied already here